### ISSUE TITLE
VS crashes after opening solution with "Cannot find service module info file '*/roslynRemoteHost.servicehub.service.json'" / Cannot execute 'discoverService'.

### ISSUE THREAD

**Issue by idontknowwhY666** [dotnet/roslyn#27781]
I could find several similar issues but not identical.

Microsoft Visual Studio Enterprise 2017 ver 15.7.3 (27703.2026)
.NET Framework ver 4.7.02558

Application: devenv.exe
Framework Version: v4.0.30319
Description: The application requested process termination through System.Environment.FailFast(string message).
Message: StreamJsonRpc.RemoteInvocationException: Request locate failed with message: Cannot find service module info file '*/roslynRemoteHost.servicehub.service.json' in 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\ServiceHub\Services', using discovery services failed: VsixServiceDiscovery: Cannot execute 'discoverService'. The underlying stream has closed.
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at StreamJsonRpc.JsonRpc.<InvokeCoreAsync>d__67`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.ThreadingTools.<WithCancellationSlow>d__11`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceHub.Client.HubClient.<LocateServiceAsync>d__40.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceHub.Client.HubClient.<RequestServiceChannelNameAsync>d__26.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.ServiceHub.Client.HubClient.<RequestServiceAsync>d__29.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.Connections.<RetryRemoteCallAsync>d__0`2.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient.Connections.<RequestServiceAsync>d__1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
Stack:
   at System.Environment.FailFast(System.String, System.Exception)
   at Microsoft.CodeAnalysis.FailFast.OnFatalException(System.Exception)
   at Microsoft.CodeAnalysis.ErrorReporting.FatalError.Report(System.Exception, System.Action`1<System.Exception>)
   at Microsoft.CodeAnalysis.ErrorReporting.FatalError.ReportUnlessCanceled(System.Exception)
   at Microsoft.VisualStudio.LanguageServices.Remote.ServiceHubRemoteHostClient+Connections+<RequestServiceAsync>d__1.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.InvokeMoveNext(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run()
   at System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction(System.Action, Boolean, System.Threading.Tasks.Task ByRef)
   at System.Threading.Tasks.Task.FinishContinuations()
   at System.Threading.Tasks.Task.FinishStageThree()
   at System.Threading.Tasks.Task`1[[System.Threading.Tasks.VoidTaskResult, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].TrySetResult(System.Threading.Tasks.VoidTaskResult)
   at System.Threading.Tasks.Task+DelayPromise.Complete()
   at System.Threading.Tasks.Task+<>c.<Delay>b__274_1(System.Object)
   at System.Threading.TimerQueueTimer.CallCallbackInContext(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.TimerQueueTimer.CallCallback()
   at System.Threading.TimerQueueTimer.Fire()
   at System.Threading.TimerQueue.FireNextTimers()
   at System.Threading.TimerQueue.AppDomainTimerCallback()

**Comment by idontknowwhY666** [comment]
roslynRemoteHost.servicehub.service.json physically in folder C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\CommonExtensions\Microsoft\ManagedLanguages\VBCSharp\LanguageServices\

{
  "host": "desktopClr.x86",
  "hostId": "RoslynCodeAnalysisService32",
  "hostGroupAllowed": true,
  "entryPoint": {
    "assemblyPath": "Microsoft.CodeAnalysis.Remote.ServiceHub.dll",
    "fullClassName": "Microsoft.CodeAnalysis.Remote.RemoteHostService",
    "appBasePath": "%VSAPPIDDIR%",
    "configPath": "%PkgDefApplicationConfigFile%"
  }
}

**Comment by idontknowwhY666** [comment]
hi jinujoseph, wouldn't be possible to fix already in 10.8? I had to 'fix' that by using older VS 15.0...

**Comment by heejaechang** [comment]
@jinujoseph we should give this bug to servicehub team.

**Comment by jinujoseph** [comment]
waiting to hear back from servicehub team, this might have already been fixed

**Comment by RyanToth3** [comment]
This issue should be fixed in an upcoming preview of Visual Studio 15.8.

**Comment by michalhainc** [comment]
In 15.8 I've started encountering this... 15.7.x was not happening

**Comment by scsloan** [comment]
Occurring for me as well after 15.8

**Comment by RyanToth3** [comment]
What's the exact error stack you're seeing? Could you run https://aka.ms/vscollect/ and share the output?

**Comment by michalhainc** [comment]
Hi, Unfortunately no I cannot, I had to reinstall asap as I'm using studio daily for my job....
I made to capture this screenshot from event viewer when it was happening to me... 

![image](https://user-images.githubusercontent.com/3962233/44302826-f0e38380-a330-11e8-825b-b7688cf94d08.png)

**Comment by neurotech** [comment]
@RyanToth3 I am using version `15.8.0` of Visual Studio Professional 2017. Here's a link to the output of `Collect.exe`:
https://www.dropbox.com/s/gjg14t8g4q10eqg/vslogs.zip?dl=0

**Comment by RyanToth3** [comment]
Would you be able to delete the cache files under %localappdata%/Microsoft/VisualStudio/15.0_<instance_id>/Extensions and then run your repro again to see if that gets you unblocked? If not could you see if you are able to install an extension through the VsixInstaller?

**Comment by neurotech** [comment]
Sorry @RyanToth3 I had to do a full Windows reset to resolve the issue as I needed VS working for my job.

**Comment by boufran1** [comment]
I am experiencing this issue when I installed the latest visual studio patch and the .NET CORE packages.

I made a zip file of the requested cache files and placed it in a google drive:

https://drive.google.com/open?id=1vQItaiiufhFvxXn8spKmK9V4XTh4W3rt

I cannot work with VS2017 anymore due to this issue.

**Comment by RyanToth3** [comment]
Could you run https://aka.ms/vscollect/ and share the output? From the cache files you shared it looks like you're issue could have a different root cause. More detailed error logs for ServiceHub issues are located in %temp%/servicehub/logs.

**Comment by boufran1** [comment]
Here is the output of the collect tool:

https://drive.google.com/file/d/188A6XqHDr7__xRGwgORAnOozA7ofde3m/view?usp=sharing

And here are the logs from %temp%/servicehub/logs:
https://drive.google.com/file/d/1UFI1zNIZvaJpHctQr6m9MKgEhncqhueb/view?usp=sharing

**Comment by jmelosegui** [comment]
Same Issue here.

https://drive.google.com/file/d/17RLgXubSQnO5S8GPDTxvl6lVQN3Ij1nZ/view?usp=sharing

**Comment by RyanToth3** [comment]
@jmelosegui I'm not seeing any service hub logs in the zip file you sent. Could you manually zip up %temp%/servicehub/logs and the *.cache files under %localappdata%/Microsoft/VisualStudio/15.0_/Extensions instead?

**Comment by jmelosegui** [comment]
Trying to debug the visual studio instance that was crashing with a new instance of visual studio I saw the following error.

```
This implementation is not part of the Windows Platform FIPS validated cryptographic algorithms.
```
Then I went to `HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Lsa`  and DELETED the `FipsAlgorithmPolicy`.

I did have the `DWORD Enabled = 0` under that subkey and it was not working. So again after deleting the subkey `FipsAlgorithmPolicy` I could open the solution again.

**Comment by boufran1** [comment]
I tried to debug as well but did not have this error, mine is still :

`Message: StreamJsonRpc.RemoteInvocationException: Request locate failed with message: Cannot find service module info file '*/roslynRemoteHost.servicehub.service.json' in 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\Common7\ServiceHub\Services', using discovery services failed: VsixServiceDiscovery: One or more errors occurred.`

Ryan, I can wait and provide additional information, logs and test potential fixes  today but I need to reformat tomorrow if I have not fixed this issue (doing a full reinstall after running the cleanup tool didn't work ). I need Visual Studio to work.

**Comment by lennoncork** [comment]
Seeing multiple developers with the same/similar issues on Windows 10 with 15.8.8. Seems to occur after windows updates are rolled out. https://developercommunity.visualstudio.com/content/problem/361164/vs-158-crashes-creating-new-console-app.html

**Comment by heejaechang** [comment]
@jinujoseph can we ping the MEF team again on this?

@lennoncork I think MEF team claimed the latest VS has a fix for it. seems that's not the case. let us check with MEF team again. thank you for the reporting.

**Comment by jinujoseph** [comment]
Issue reported to MEF at [link](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/716931)

**Comment by heejaechang** [comment]
@dgriffen okay, I probably need to say it is extension manager bug, rather than broadly saying everything as MEF issue. 

I believe VsixDiscoveryService is using ExtensionManager API to read servicehub json files defined in vsix manifest and for various reasons (MEF cache corrupted, json file in extension folder got locked up, registry entries where extension related info supposed to be saved is messed up, or locked by other process and etc), it can't read that file so service hub fails to provide the service which lead IDE feature to fail.

fixing root cause would be the best, but if that is not easy, can we at least try to recover (most of time, fixing MEF cache fix the issue), or have a way to tell users how to fix recover?

**Comment by dgriffen** [comment]
MEF happens outside the extension manager and only operates in devenv so this wouldn't fix the issue. The problem this particular user has is that somehow their private registry is not allowing servicehub to create new keys. We're trying to get procmon traces from them now to try and determine what is going on.

**Comment by emojean1** [comment]
Same problem here with Visual Studio 2017 V15.9.11

**Comment by adrachnev** [comment]
> Would you be able to delete the cache files under %localappdata%/Microsoft/VisualStudio/15.0_<instance_id>/Extensions and then run your repro again to see if that gets you unblocked? If not could you see if you are able to install an extension through the VsixInstaller?


Have same issue with VS 2019 Pro starting VS as admin. Starting devenv from command line seems fine.


[extensions.de-DE.zip](https://github.com/dotnet/roslyn/files/3926130/extensions.de-DE.zip)

**Comment by LordDelacroix** [comment]
well, yippie - this beastly problem has surfaced for me on 16.9.31313.79 and has been for several weeks now. The only resolution I've been able to find is to repair the entire installation and if I install ANY extensions, this problem surfaces.

I get these 3 lines in the service hub logs folders latest log file after I kill the VS process that is hung.

05/19/2021 15:07:42 Mountain Standard Time : Error : Vsix service path 'C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\RAZORLANGUAGESERVICES\Microsoft.VisualStudio.Razor.TagHelperProvider.servicehub.service' file name does not end with '.servicehub.service.json'
05/19/2021 15:07:42 Mountain Standard Time : Error : Vsix service path 'C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\RAZORLANGUAGESERVICES\Microsoft.VisualStudio.Razor.TagHelperProvider64.servicehub.service' file name does not end with '.servicehub.service.json'
05/19/2021 15:07:42 Mountain Standard Time : Error : Vsix service path 'C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2019\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\RAZORLANGUAGESERVICES\Microsoft.VisualStudio.Razor.TagHelperProvider64S.servicehub.service' file name does not end with '.servicehub.service.json'

**Comment by paulneibarger** [comment]
This started for me today with version 16.11.10? Is Uninstall/Install the only solution?

**Comment by AddioElectronics** [comment]
I'm having a very similar issue in an experimental instance.
I renamed the hive and now 3 services aren't able to initialize, all referencing the same DLL `"assemblyPath": ".\\Microsoft.CodeAnalysis.Remote.ServiceHub.dll",`
Before renaming the hive these errors were not being displayed.

These are the JSONs.
```
Microsoft.VisualStudio.LanguageServices.AsynchronousOperationListenerCore64.servicehub.service.json (ErrorKind: Error HResult: 80070002)
Microsoft.VisualStudio.LanguageServices.LegacySolutionEventsAggregationCore64.servicehub.service.json (ErrorKind: Error HResult: 80070002)
Microsoft.VisualStudio.LanguageServices.DesignerAttributeDiscoveryCore64.servicehub.service.json (ErrorKind: Error HResult: 80070002)
```
